app/src/main/kotlin/github/com/rhacco/dota2androidapp/activities/LiveMatchesActivity.kt

```diff
@@ -38,7 +42,20 @@ class LiveMatchesActivity : BaseLifecycleActivity<MatchesViewModel>() {
             }
         })
         mViewModel.mLiveMatchesItemDataQuery.observe(this, Observer<LiveMatchesItemData> {
-            it?.let { itemData -> mAdapter.add(itemData) }
+            it?.let { itemData ->
+                if (mAdapter.add(itemData))
+                    recycler_view.layoutManager.scrollToPosition(0)
+                else
+                    mViewModel.checkIfMatchFinished(itemData.mMatchID)
+            }
+        })
+        mViewModel.mMatchFinishedQuery.observe(this, Observer<Pair<Long, Boolean>> {
+            it?.let { (matchId, isFinished) ->
+                if (isFinished) {
+                    mAdapter.remove(matchId)
+                    mViewModel.removeFinishedMatch(matchId)
```

Comment:
Having to do the deletion on both the adapter and the viewmodel defeats the purpose of the viewmodel providing the updates. 

I would suggest just calling .removeFinishedMatch and opening a channel on the viewmodel to listen to when the matchId is removed. Once that occurs, then call mAdapter.remove().

Reply:
Done, thanks